pypesto/select/method.py

```diff
@@ -0,0 +1,512 @@
+"""Functionality related to using a PEtab Select model selection method."""
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import petab_select
+from petab_select import (
+    VIRTUAL_INITIAL_MODEL,
+    CandidateSpace,
+    Criterion,
+    Method,
+    Model,
+)
+
+from ..C import TYPE_POSTPROCESSOR
+from .model_problem import ModelProblem
+
+
+class MethodSignalProceed(str, Enum):
+    """Indicators for how a model selection method should proceed."""
+
+    # TODO move to PEtab Select?
+    STOP = 'stop'
+    CONTINUE = 'continue'
+
+
+@dataclass
+class MethodSignal:
+    """The state of a model selection method after a single model calibration.
+
+    Attributes
+    ----------
+    accept:
+        Whether to accept the model.
+    proceed:
+        How the method should proceed.
+    """
+
+    accept: bool
+    # TODO change to bool?
+    proceed: MethodSignalProceed
+
+
+class MethodLogger:
+    """Log results from a model selection method.
+
+    Attributes
+    ----------
+    level:
+        The logging level.
+    logger:
+        A logger from the `logging` module.
+    """
+
+    def __init__(
+        self,
+        level: str = 'info'
+    ):
+        self.logger = logging.getLogger(__name__)
+        self.level = level
+
+    def log(self, message, level: str = None) -> None:
+        """Log a message.
+
+        Parameters
+        ----------
+        message:
+            The message.
+        level:
+            The logging level. Defaults to the value defined in the
+            constructor.
+        """
+        if level is None:
+            level = self.level
+        getattr(self.logger, level)(message)
+
+    def new_selection(self) -> None:
+        """Start logging a new model selection."""
+        padding = 20
```

Comment:
to improve readability, you could try having the whole thing as a pandas.DataFrame and then just logging that (or a subset of columns)

Reply:
e.g.:
```python
import pandas as pd
import logging

logger = logging.getLogger()

df = pd.DataFrame({'column a': [1,2,3], column 'b': [121,122,123]})
logger.warning(df)
```

Reply:
Switched to fixed-length strings